src/lib/fetch.ts

```diff
@@ -174,6 +175,25 @@ export function _sessionResponse(data: any): AuthResponse {
   return { data: { session, user }, error: null }
 }
 
+export function _sessionResponsePassword(data: any): AuthResponsePassword {
+  const response = _sessionResponse(data) as AuthResponsePassword
+
+  if (
+    !response.error &&
+    data.weak_password &&
+    typeof data.weak_password === 'object' &&
```

Comment:
if possible, consider assigning semantic meaning to parts of this check. It's a little hard to read at first glance.

Reply:
Yeah will probably remove this validation in a follow up PR, doesn't make much sense.